module/x/gravity/keeper/keeper.go

```diff
@@ -562,3 +562,65 @@ func (k Keeper) CreateContractCallTx(ctx sdk.Context, invalidationNonce uint64,
 	)
 	return newContractCallTx
 }
+
+/////////////////
+// MIGRATE     //
+/////////////////
+
+// Clean up all state associated a previous gravity contract and set a new contract. This is intended to run in the upgrade handler.
+func (k Keeper) MigrateGravityContract(ctx sdk.Context, newBridgeAddress string, bridgeDeploymentHeight uint64) {
```

Comment:
We might want to note that this function doesn't move over any of the ETH based state to the new gravity contract. This is only for the case where there is nothing that needs to be moved over on the ETH side of things. We would likely need separate codes path and flows for:
1. Freezing the gravity contract via governance
2. Requesting the balance of all ERC20 tokens in the gravity contract to be sent via the ETH oracle and verifying with cosmos side supply
3. Outgoing tx to send the ERC20 tokens to the new gravity contract from the old one
That would be in addition to this function which clears state. We might want to add some comments to this effect.